modules/miner/cpuminer.go

```diff
@@ -2,11 +2,18 @@ package miner
 
 import (
 	"time"
+
+	"github.com/NebulousLabs/Sia/build"
 )
 
 // threadedMine starts a gothread that does CPU mining. threadedMine is the
 // only function that should be setting the mining flag to true.
 func (m *Miner) threadedMine() {
+	if err := m.tg.Add(); err != nil {
+		return
+	}
+	defer m.tg.Done()
+
```

Comment:
There's a second bit of trickiness related to this function, and I'm not sure that the testing coverage is complete enough to detect problems. I'm pretty sure that if you were to call `Close` right now while the cpu miner was running, it'd never actually shut down. You should write a test to verify this.

`Stop` cannot return until `m.tg.Done()` has returned, which won't happen until after this function terminates. And this function won't terminate if `m.miningOn`. The simplest fix is to change the mining statement to `if m.mining || !m.miningOn || m.tg.IsStopped() {`, and then you'll also want to add the line `m.miningOn = false` to the `Close` function.